src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/RedundantImportCheck.java

```diff
@@ -141,6 +149,31 @@ else if (pkgName != null && isFromPackage(imp.getText(), pkgName)) {
         }
     }
 
+    @Override
+    public void finishTree(DetailAST ast) {
+
+        // check for types already imported by star imports
+        // we should run this only if there is only 1 star import in file
+        // 0 star imports -- there will be no violations
+        // > 1 star imports -- false positives are possible
+        if (starImports.size() == 1) {
+            final FullIdent starImp = starImports.get(0);
+
+            imports.stream()
+                .filter(imp -> !isStarImport(imp))
```

Comment:
Is there a reason `imports` need to also contain star imports?
Why can't line 133 be the else of line 135's if?

Reply:
Currently it is used to spot duplicate star imports. But this can be implemented other way